jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -369,7 +369,9 @@ private static boolean isSamePackage(Class<?> a, Class<?> b){
 		}
 		
 		void checkAccess(MethodHandle handle, boolean skipAccessCheckPara) throws IllegalAccessException {
-			if (INTERNAL_PRIVILEGED == accessMode) {
+			if ((INTERNAL_PRIVILEGED == accessMode)
+				|| (FULL_ACCESS_MASK == accessMode)
```

Comment:
Seems adding this is causing a number of test suite failures, even in jdk11.
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.functional_s390x_linux_Personal/413

Reply:
I added this line because OpenJ9 is throwing IllegalAccessException with message "K0587". The new Java 15 code from InnerClassLambdaMetafactory is passing in accessMode FULL_ACCESS_MASK. It is calling from findConstructor().

Reply:
OK, I seems to know what is going on. definingClass should be made nestmate of accessClass which is missing here.

Reply:
I have removed the checks for FULL_ACCESS_MASK. 

I don't find an existing APIs (Java or native) that could let me make a class to be nestmate of another class. So I have changed the native implementation of getNestHost() to return the nesthost of the loopup class for hidden classes (which is defined by Unsafe.defineAnonyousClass()). 

I am able to build Java 15 with the current change.